docs/architecture/maui/unit-testing.md

```diff
@@ -1,9 +1,223 @@
+
+> [!NOTE]
+> Unit tests are very effective against regression. That is, functionality that used to work, but has been disturbed by a faulty update.
+
+Unit tests typically use the arrange-act-assert pattern:
+
+| Step | Description |
+|---------|---------|
+| Arrange | Initializes objects and sets the value of the data that is passed to the method under test. |
+| Act | Invokes the method under test with the required arguments. |
+| Assert | Verifies that the action of the method under test behaves as expected. |
+
+This pattern ensures that unit tests are readable, self-describing, and consistent.
+
+## Dependency injection and unit testing
+
+One of the motivations for adopting a loosely-coupled architecture is that it facilitates unit testing. One of the types registered with the dependency injection service is the `IAppEnvironmentService` interface. The following code example shows an outline of this class:
+
+```csharp
+public class OrderDetailViewModel : ViewModelBase
+{
+    private IOrderService _ordersService;
+
+    public OrderDetailViewModel(
+        IAppEnvironmentService appEnvironmentService,
+        IDialogService dialogService, INavigationService navigationService, ISettingsService settingsService)
+        : base(dialogService, navigationService, settingsService)
+    {
+        _appEnvironmentService = appEnvironmentService;
+        _settingsService = settingsService;
+    }
+}
+```
+
+The `OrderDetailViewModel` class has a dependency on the `IAppEnvironmentService` type, which the dependency injection container resolves when it instantiates an `OrderDetailViewModel` object. However, rather than create an `IAppEnvironmentService` object which utilizes real servers, devices and configurations to unit test the `OrderDetailViewModel` class, instead, replace the `IAppEnvironmentService` object with a mock for the purpose of the tests.
+
+This approach allows the `IAppEnvironmentService` object to be passed into the `OrderDetailViewModel` class at runtime, and in the interests of testability, it allows a mock class to be passed into the `OrderDetailViewModel` class at test time. The main advantage of this approach is that it enables unit tests to be executed without requiring unwieldy resources such as runtime platform features, web services, or databases.
+
+## Testing MVVM applications
+
+Testing models and view models from MVVM applications is identical to testing any other classes, and uses the same tools and techniques; this includes features such as unit testing and mocking. However, some patterns that are typical to model and view model classes can benefit from specific unit testing techniques.
+
+> [!TIP]
+> Test one thing with each unit test.
```

Comment:
Why—I mean, I know why...but your reader will be asking themselves, why. Add the reason why as part of this statement.

Reply:
Expanded and added a link for additional best practices.